### ISSUE TITLE
[Enhancement] Allow using chorus fruit to teleport from "allowed territory" to "not allowed territory"

### ISSUE THREAD

**Issue by kain888** [magnusulf/Factions#102]
## Affected Plugin

- **Affected Plugin Name:** Factions
- **Affected Plugin Version:** 3.3.3

## Current Situation
Chorus fruit is food that may teleport the player up to 8 blocks in any direction after eating it. Currently, to prevent entering other factions' area too easily and forgivingly, the plugin allows for disabling chorus fruit teleportation in various territories as expressed in the following lines in the config:
-   "allowChorusFruitInEnemyTerritory":
-   "allowChorusFruitInWildernessTerritory":
-   "allowChorusFruitInOwnTerritory":
-   "allowChorusFruitInOtherTerritory":

The main reason the Factions servers' disable it, is that is's too powerful effect with too little cost to achieve, thus overpowered.

**It was requested here:** https://github.com/magnusulf/Factions/issues/51

The plugin blocks you from teleporting to not allowed territory even when you are using it on the allowed territory. For example, standing on my own claimed territory will not allow me to teleport to places where I'm not allowed to teleport in.

## Proposed Change

Add ability to allow teleporting **_into not allowed territory_** while _**standing on the territory you are allowed to**_ use chorus fruit on. 

**Example**

With config:
"allowChorusFruitInOwnTerritory": true
"allowChorusFruitInEnemyTerritory": false

AND 

standing in own chunk border that is next to enemy chunk;

it would allow you to both use chorus fruit and have a chance to successfully teleport into enemy territory. After teleporting to enemy territory you would be blocked to use chorus fruit, because "allowChorusFruitInEnemyTerritory" is set to false.

Basically, it would make borders more easy to penetrate, but also would make people who teleport in more likely to get stuck on the enemy territory. This would make the closest 8 blocks of bordering territory more dangerous to occupy.

## Reasons for Change

The ability to teleport 8 blocks within borders would open an array of new tactics and unleash a creativity in how to prevent them. It would sometimes unlock a high-risk, high-reward move in a warfare.

The current solution to overpowered chorus fruit, unfortunately, leaves the unique effect of chorus fruit totally nerfed, nullifying any use of it. The proposed change would bring the fun from chorus fruit back to Minecraft Factions, but controlling its power in a more balanced way.
